src/zenml/new/steps/step_context.py

```diff
@@ -177,18 +185,34 @@ def model(self) -> "Model":
         Raises:
             StepContextError: If the `Model` object is not set in `@step` or `@pipeline`.
         """
+        shall_warm_up = True
+        step_run = None
         if self.step_run.config.model is not None:
-            model = self.step_run.config.model
+            if self.step_run.model_version:
+                model = self.step_run.model_version.to_model_class()
+                shall_warm_up = False
+            else:
+                model = self.step_run.config.model
+                step_run = self.step_run
         elif self.pipeline_run.config.model is not None:
-            model = self.pipeline_run.config.model
+            if self.pipeline_run.model_version:
+                model = self.pipeline_run.model_version.to_model_class()
+                shall_warm_up = False
+            else:
+                model = self.pipeline_run.config.model
         else:
             raise StepContextError(
                 f"Unable to get Model in step `{self.step_name}` of pipeline "
                 f"run '{self.pipeline_run.id}': it was not set in `@step` or `@pipeline`."
             )
 
         # warm-up the model version
-        model._get_or_create_model_version()
+        if shall_warm_up:
```

Comment:
Why does this need to happen here? Can't we use e.g. code in the step launcher to always load/create the model version that is configured for a step before the step even launches, which means in all other places we can simply assume if the `StepRunResponse` has no model associated with it, there is none?

Reply:
Addressed